lib/collection/cookie.js

```diff
@@ -269,9 +269,14 @@ _.assign(Cookie.prototype, /** @lends Cookie.prototype */ {
             str += '; HttpOnly';
         }
 
-        if (this.extensions) {
+        if (_.isArray(this.extensions)) {
             this.extensions.forEach(function (ext) {
-                str += '; ' + ext;
+                if (_.isNil(ext.key)) {
+                    return;
+                }
+
+                var value = (ext.value) ? ext.value : true;
+                str += '; ' + ext.key + '=' + value;
```

Comment:
Nitpick: You can also use a template string for better readability.
```suggestion
                str += `; ${ext.key}=${value}`;
```

Just a personal preference. Feel free to ignore. 🙃